pkgs/tools/security/munge/default.nix

```diff
@@ -1,42 +1,73 @@
-{ lib, stdenv, fetchFromGitHub, autoreconfHook, libgcrypt, zlib, bzip2 }:
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  autoreconfHook,
+  libgcrypt,
+  zlib,
+  bzip2,
+}:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "munge";
   version = "0.5.16";
 
   src = fetchFromGitHub {
     owner = "dun";
     repo = "munge";
-    rev = "${pname}-${version}";
+    rev = "munge-${finalAttrs.version}";
     sha256 = "sha256-fv42RMUAP8Os33/iHXr70i5Pt2JWZK71DN5vFI3q7Ak=";
   };
 
-  strictDeps = true;
   nativeBuildInputs = [
     autoreconfHook
     libgcrypt # provides libgcrypt.m4
   ];
-  buildInputs = [ libgcrypt zlib bzip2 ];
 
-  preAutoreconf = ''
-    # Remove the install-data stuff, since it tries to write to /var
-    substituteInPlace src/Makefile.am --replace "etc \\" "\\"
-  '';
+  buildInputs = [
+    libgcrypt
+    zlib
+    bzip2
+  ];
+
+  strictDeps = true;
 
   configureFlags = [
+    # Load data from proper global paths
     "--localstatedir=/var"
+    "--sysconfdir=/etc"
+    "--runstatedir=/run"
+    "--with-sysconfigdir=/etc/default"
+
+    # Install data to proper directories
+    "--with-pkgconfigdir=${placeholder "out"}/lib/pkgconfig"
+    "--with-systemdunitdir=${placeholder "out"}/lib/systemd/system"
+
+    # Cross-compilation fixes
     "--with-libgcrypt-prefix=${libgcrypt.dev}"
```

Comment:
```suggestion
    "--with-libgcrypt-prefix=${lib.getDev libgcrypt}"
```

Reply:
I am not even sure if this is still necessary. \<rant\>Cross people keep introducing junk like this without any mention how to reproduce the failure or attempts to resolve this upstream. I am half inclined to just revert <https://github.com/NixOS/nixpkgs/commit/faacc88c93086982e2c95708176863f415e03810> @NickCao\</rant\>

Reply:
+1 to `getDev`, but I'm not sure I understand what issue the comment refers to (even though I suspect I've seen it before)

Reply:
Okay, done.

By the way, I managed to reproduce a cross failure with `pkgsCross.aarch64-multiplatform.munge` when I revert https://github.com/NixOS/nixpkgs/commit/faacc88c93086982e2c95708176863f415e03810.

The `libgcrypt.dev` in `nativeBuildInputs` is needed for `libgcrypt.m4` to be picked up for `ACLOCAL_PATH` (see also https://github.com/NixOS/nixpkgs/pull/311773#discussion_r1612536323). And `--with-libgcrypt-prefix` is probably needed because the m4 file/`libgcrypt-config` cannot handle multiple outputs.

It appears possible to force it to use `pkg-config` but that is no less hacky:

<details>
<summary>Patch</summary>


```patch
diff --git a/pkgs/tools/security/munge/default.nix b/pkgs/tools/security/munge/default.nix
index f21a9e17add3..1a7ada38fa95 100644
--- a/pkgs/tools/security/munge/default.nix
+++ b/pkgs/tools/security/munge/default.nix
@@ -1,6 +1,7 @@
 {
   lib,
   stdenv,
+  pkg-config,
   fetchFromGitHub,
   autoreconfHook,
   libgcrypt,
@@ -21,7 +22,7 @@ stdenv.mkDerivation (finalAttrs: {
 
   nativeBuildInputs = [
     autoreconfHook
-    libgcrypt # provides libgcrypt.m4
+    pkg-config
   ];
 
   buildInputs = [
@@ -44,7 +45,7 @@ stdenv.mkDerivation (finalAttrs: {
     "--with-systemdunitdir=${placeholder "out"}/lib/systemd/system"
 
     # Cross-compilation hacks
-    "--with-libgcrypt-prefix=${lib.getDev libgcrypt}"
+    "GPGRT_CONFIG=${stdenv.cc.targetPrefix}pkg-config"
     # workaround for cross compilation: https://github.com/dun/munge/issues/103
     "ac_cv_file__dev_spx=no"
     "x_ac_cv_check_fifo_recvfd=no"
@@ -57,6 +58,11 @@ stdenv.mkDerivation (finalAttrs: {
     "sysconfigdir=${placeholder "out"}/etc/default"
   ];
 
+  preAutoreconf = ''
+    # For some reason, automake setup hook does not pick it up from buildInputs.
+    addAclocals "${lib.getDev libgcrypt}"
+  '';
+
   postInstall = ''
     # rmdir will notify us if anything new is installed to the directories.
     rmdir "$out"/{var{/{lib,log}{/munge,},},etc/munge}
```

</details>

Ideally, all of this would be resolved upstream but `m4` is the opposite of exciting to work with.